src/module/data/item/base.mjs

```diff
@@ -78,9 +79,13 @@ export default class BaseItemModel extends SubtypeModelMixin(foundry.abstract.Ty
 
   /** @inheritdoc */
   async toEmbed(config, options = {}) {
-
-    options.rollData ??= this.parent.getRollData();
-    const enriched = await foundry.applications.ux.TextEditor.implementation.enrichHTML(this.description.value, options);
+    // Override document-related options with this document's info so it's consistent wherever embedded
+    const enriched = await enrichHTML(this.description.value, {
+      ...options,
+      relativeTo: this.parent,
+      rollData: this.parent.getRollData(),
+      secrets: this.parent.isOwner,
```

Comment:
Roll data and secrets are done in the helper method. imo we dont have a reason to enforce and override what is passed into `options`